test/phoenix_ecto/plugs/check_repo_ready_test.exs

```diff
@@ -0,0 +1,214 @@
+defmodule Phoenix.Ecto.Plugs.CheckRepoReadyTest do
+  use ExUnit.Case, async: true
+  use Plug.Test
+
+  alias Phoenix.Ecto.Plugs.CheckRepoReady
+
+  defmodule LongLivedProcess do
+    def run do
+      Process.sleep(1_000)
+      run()
+    end
+  end
+
+  defmodule StorageUpRepo do
+    defmodule Adapter, do: def(storage_status(_repo_config), do: :up)
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  defmodule StorageDownRepo do
+    defmodule Adapter, do: def(storage_status(_repo_config), do: :down)
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  defmodule NoStorageStatusRepo do
+    defmodule Adapter, do: nil
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  defmodule StorageErrorRepo do
+    defmodule Adapter,
+      do: def(storage_status(_repo_config), do: {:error, %{message: "Adapter error"}})
+
+    def config, do: []
+    def __adapter__, do: Adapter
+  end
+
+  test "does not raise an error when the storage is created and there are no pending migrations" do
+    Process.register(self(), StorageUpRepo)
+    Application.put_env(:check_repo_ready, :ecto_repos, [StorageUpRepo])
+    get_migrations_function = fn _repo -> [] end
+
+    conn = conn(:get, "/")
+
+    assert conn ==
+             CheckRepoReady.call(
+               conn,
+               otp_app: :check_repo_ready,
+               get_migrations_function: get_migrations_function
+             )
+  after
+    Application.delete_env(:check_repo_ready, :ecto_repos)
+    Process.unregister(StorageUpRepo)
```

Comment:
It is not necessary to unregister because any name is removed when the process dies.

Reply:
IIRC it's one process per file so the process dies only after the whole test run is finished, right? 
So this could cause an error when we try to register another process with the same name.
Or I'm remembering it wrong? 

Actually, I _had_ to unregister all the processes after the test because _sometimes_ the test failed with a `(ArgumentError) could not register #PID<> with the name Repo because it is not alive, the`**`name is already taken`**`...`, and I think that wouldn't be nice to have intermittent tests.

If I remove all the calls to `Process.unregister` in the after block and run the tests with a specific seed they fail, but If I run with another seed than it passes.

<details>
  <summary> Tests passing and failing </summary>
  <img src="https://user-images.githubusercontent.com/14015177/68482777-0afcd080-0219-11ea-8bd8-6ec278c9b110.png">
</details>

Maybe I'm missing something and the problem is actually another thing. 🤷‍♂

Reply:
Gah, you are correct. The clean up of Process.unregister is asynchronous, which means if the next test starts and tries to use the same name, then there is a race. Let's leave the unregister calls in. Good call!

Reply:
Now I know why, tks! 💙